core/src/apps/cardano/sign_tx.py

```diff
@@ -40,62 +50,38 @@ def is_change(output, inputs):
 
 
 async def show_tx(
-    ctx,
-    outputs: list,
-    outcoins: list,
+    ctx: wire.Context,
+    inputs: List[CardanoTxInputType],
+    outputs: List[CardanoTxOutputType],
     fee: int,
     network_name: str,
-    raw_inputs: list,
-    raw_outputs: list,
 ) -> None:
+    total_amount = 0
     for index, output in enumerate(outputs):
-        if is_change(raw_outputs[index].address_n, raw_inputs):
+        if is_change(output.address_n, inputs):
             continue
 
-        await confirm_sending(ctx, outcoins[index], output)
-
-    total_amount = sum(outcoins)
-    await confirm_transaction(ctx, total_amount, fee, network_name)
+        total_amount += output.amount
+        await confirm_sending(ctx, output.amount, output.address)
 
+    if total_amount > LOVELACE_INVALID:
+        raise ValueError("Total transaction amount is out of range")
```

Comment:
Isn't this a wire error, caused by user inputs?

Reply:
Makes sense.